### ISSUE TITLE
 How to import the LM-O dataset 

### ISSUE THREAD

**Issue by flashmoment** [aau-cns/poet#18]
How to import the LM-O dataset into the project? I downloaded the PBR dataset and models for LM-O from the BOP website, as well as the BOP19-23 test image set. How should I import them, and what is the project structure like? Can you provide a sample for me to refer to?

**Comment by z7r7y7** [comment]
Hello, you can take a look at the poet/data_utils/data_annotation/lmo2poet.py file, which generates annotation files for loading the dataset.

**Comment by flashmoment** [comment]
> Hello, you can take a look at the poet/data_utils/data_annotation/lmo2poet.py file, which generates annotation files for loading the dataset.

Thank you. I have reviewed the the poet/data_utils/data_annotation/lmo2poet.py file. After placing the LMO dataset into the project, I ran lmo2poet.py, and it generated two .json files - one named train.json and the other train_synt.json. I've also examined the contents of these JSON files, which contain specific information about the dataset. However, I'm unsure about their significance and how the network will extract data from my dataset.

The PBR dataset includes RGB images, depth, mask files for 15 models. I'm uncertain if the code will seamlessly extract the files for the 8 models used in the LMO dataset from these files. I'm also unsure about how the network retrieves the dataset from the dataset folder and from which path it locates the dataset folder. Do I need to manually modify the downloaded PBR dataset, or should I directly include it in the project

**Comment by tgjantos** [comment]
Hi!

As you already did, and how it described in the [README](https://github.com/aau-cns/poet?tab=readme-ov-file#bop-datasets), you can process the BOP datasets into JSON files that contain the information about the dataset. The JSON files contains the path to each image and the corresponding annotations (relative pose, class bbox). If you provide the path to this file the dataloader will correctly load the data for the network to train/evaluate. 

Regarding the 3D object model: BOP datasets usually provide two sets of 3D models. The full model and a sampled one for faster evaluation. You can simply provide this information to the network through the runtime argument [--models](https://github.com/aau-cns/poet/blob/main/main.py#L142).

I hope this helps you!

Best,
Thomas

**Comment by flashmoment** [comment]
> Hi!
> 
> As you already did, and how it described in the [README](https://github.com/aau-cns/poet?tab=readme-ov-file#bop-datasets), you can process the BOP datasets into JSON files that contain the information about the dataset. The JSON files contains the path to each image and the corresponding annotations (relative pose, class bbox). If you provide the path to this file the dataloader will correctly load the data for the network to train/evaluate.
> 
> Regarding the 3D object model: BOP datasets usually provide two sets of 3D models. The full model and a sampled one for faster evaluation. You can simply provide this information to the network through the runtime argument [--models](https://github.com/aau-cns/poet/blob/main/main.py#L142).
> 
> I hope this helps you!
> 
> Best, Thomas

Thank you for your guidance. I have another question, this time about the dataset. I noticed in the lmo2poet file, there's a code snippet data_paths = ['train_synt/', 'train_pbr/'], which requires the paths for two datasets. I downloaded the PBR dataset from the BOP website by clicking on "PBR-BlenderProc4BOP training images," which contains 50,000 RGB images. This should be the train_pbr dataset used by your team in experiments. However, what should I provide for the 'train_synt/' dataset path?

 Since your team did not use synthetic images for training, but I noticed an option on the BOP website under the LMO dataset, labeled as "Synt. training images of isolated objects." Should I download this dataset as well? This way, I can obtain a train.json file along with a train_synt file.

 Then, I will apply the path of the train.json file to parser.add_argument('--train_set', default="train", type=str, help="Determine on which dataset split to train"). As for the test set, do I need to download the BOP'19-23 test images dataset from the BOP website, place it in the lmo2poet file, and regenerate a JSON file? Then, I'll apply the path of this JSON file to parser.add_argument('--eval_set', default="test", type=str, help="Determine on which dataset split to evaluate"). Finally, adjust the --n_classes parameter, and the training will commence, correct?

**Comment by flashmoment** [comment]
> Hi!
> 
> As you already did, and how it described in the [README](https://github.com/aau-cns/poet?tab=readme-ov-file#bop-datasets), you can process the BOP datasets into JSON files that contain the information about the dataset. The JSON files contains the path to each image and the corresponding annotations (relative pose, class bbox). If you provide the path to this file the dataloader will correctly load the data for the network to train/evaluate.
> 
> Regarding the 3D object model: BOP datasets usually provide two sets of 3D models. The full model and a sampled one for faster evaluation. You can simply provide this information to the network through the runtime argument [--models](https://github.com/aau-cns/poet/blob/main/main.py#L142).
> 
> I hope this helps you!
> 
> Best, Thomas

 parser.add_argument('--class_info', type=str, default='/annotations/classes.json',
                        help='path to .txt-file containing the class names')
    parser.add_argument('--models', type=str, default='/models_eval/',
                        help='path to a directory containing the classes models')
    parser.add_argument('--model_symmetry', type=str, default='/annotations/symmetries.json',
                        help='path to .json-file containing the class symmetries')

n main.py, I only found two places where JSON files are required. The first one should be using the lmo_classes.json file provided by you in the dataset_files folder. The second one should be using the lmo_symmetries.json file also provided by you in the dataset_files folder. What should I do with the JSON files I generated myself?

**Comment by tgjantos** [comment]
The JSON files generated should be placed into an annotations folder that matches the structure required. If you check out the [dataset builder function](https://github.com/aau-cns/poet/blob/main/data_utils/pose_dataset.py#L320) you will see that it will look for the annotation file in `root/annotations/train.json`, where `root` refers to the top level of the data folder.

Hope this helps you!

best,
Thomas

**Comment by flashmoment** [comment]
> The JSON files generated should be placed into an annotations folder that matches the structure required. If you check out the [dataset builder function](https://github.com/aau-cns/poet/blob/main/data_utils/pose_dataset.py#L320) you will see that it will look for the annotation file in `root/annotations/train.json`, where `root` refers to the top level of the data folder.
> 
> Hope this helps you!
> 
> best, Thomas

thank you very much

**Comment by tgjantos** [comment]
The keyframe.txt file contains the images that are considered as the keyframes for the BOP challenge. You have to use this file to generate the json annotation files. Hope this helps!

Best,
Thomas

**Comment by Fusica** [comment]
> The keyframe.txt file contains the images that are considered as the keyframes for the BOP challenge. You have to use this file to generate the json annotation files. Hope this helps!
> 
> Best, Thomas

Is this for use with the YCBV dataset or the LMO dataset? I successfully ran eval and train without generating keyframe.json, could there be any issues?
